src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,24 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Transaction transaction = null;
+        try (Session session = factory.openSession()) {
+            transaction = session.beginTransaction();
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add actor to DB", e);
+        }
+        return actor;
```

Comment:
it seems that implementations for `ActorDao`, `MovieDao` and `CountryDao` are identical. therefore it may be considered as a violation of DRY

you could create a method in abstract parent:
```
protected <T> void doAdd(T entity) {
}
```
and use that logic in this method. this way you would just call that method in each implementation so that the code isn't duplicated

however I'm not sure whether you're allowed to modify `AbstactDao` - you might ask the question in our Slack chat

if you're now allowed - feel free to skip this comment :)

Reply:
Good idea, I'm not sure exactly if we can change AbstactDao, but why not :)